SU2_CFD/src/fluid/CFluidFlamelet.cpp

```diff
@@ -0,0 +1,245 @@
+  string name_prog = table_scalar_names[I_PROGVAR];
+
+  /*--- value for the progress variable and enthalpy ---*/
+  su2double enth = val_scalars[I_ENTH];
+  su2double prog = val_scalars[I_PROGVAR];
+
+  /*--- perform table lookup ---*/
+  unsigned long exit_code = look_up_table->LookUp_XY(varnames_Sources, val_vars_Sources, prog, enth);
+
+  /*--- the source term for the progress variable is always positive by construction, but we clip from below  --- */
+  source_scalar[I_PROGVAR] = max(EPS, table_sources[I_SRC_TOT_PROGVAR]);
+  source_scalar[I_ENTH] = 0.0;
+
+  /*--- source term for the auxiliary species transport equations---*/
+  for (size_t i_aux = 0; i_aux < n_user_scalars; i_aux++) {
+    /*--- The source term for the auxiliary equations consists of a production term and a consumption term:
+          S_TOT = S_PROD + S_CONS * Y ---*/
+    su2double y_aux = val_scalars[n_control_vars + i_aux];
+    su2double source_prod = table_sources[1 + 2 * i_aux];
+    su2double source_cons = table_sources[1 + 2 * i_aux + 1];
+    source_scalar[n_control_vars + i_aux] = source_prod + source_cons * y_aux;
+  }
+
+  return exit_code;
+}
+
+void CFluidFlamelet::SetTDState_T(su2double val_temperature, const su2double* val_scalars) {
+  su2double val_enth = val_scalars[I_ENTH];
+  su2double val_prog = val_scalars[I_PROGVAR];
+
+  string name_enth = table_scalar_names[I_ENTH];
+  string name_prog = table_scalar_names[I_PROGVAR];
+
+  /*--- add all quantities and their address to the look up vectors ---*/
+  look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, val_enth);
+
+  /*--- compute Cv from Cp and molar weight of the mixture (ideal gas) ---*/
+  Cv = Cp - UNIVERSAL_GAS_CONSTANT / molar_weight;
+}
+
+/* --- Total enthalpy is the transported variable, but we usually have temperature as a boundary condition,
+       so we do a reverse lookup */
+unsigned long CFluidFlamelet::GetEnthFromTemp(su2double* val_enth, const su2double val_prog, const su2double val_temp,
+                                              su2double initial_value) {
+  string name_prog = table_scalar_names[I_PROGVAR];
+  string name_enth = table_scalar_names[I_ENTH];
+
+  /* convergence criterion for temperature in [K], high accuracy needed for restarts. */
+  su2double delta_temp_final = 0.001;
+  su2double enth_iter = initial_value;
+  su2double delta_enth;
+  su2double delta_temp_iter = 1e10;
+  unsigned long exit_code = 0;
+  vector<string> look_up_tags;
+  vector<su2double*> look_up_data;
+  int counter_limit = 1000;
+
+  int counter = 0;
+  while ((abs(delta_temp_iter) > delta_temp_final) && (counter++ < counter_limit)) {
+    /* look up temperature and heat capacity */
+    look_up_table->LookUp_XY(varnames_TD, val_vars_TD, val_prog, enth_iter);
+
+    /*--- calculate delta_temperature ---*/
+    delta_temp_iter = val_temp - Temperature;
+
+    /* calculate delta_enthalpy following dh = cp * dT */
+    delta_enth = Cp * delta_temp_iter;
```

Comment:
I see why you are doing it but from a code-readability/safety perspective it would be better to pass a vector of plain su2double to the table (not pointer) and then have a helper function to unpack the vector into the different variables.

Reply:
OK, my interpretation is: 
vector<su2double*> val_vars_TD will become vector<su2double> val_vars_TD
and after a call to lookup_xy, that puts the values in val_vars_TD, I do this:
`    Temperature = val_vars_TD[GetIdx(varnames_TD,"Temperature")];`

Reply:
I think that is less surprising when reading the code :)